src/Entity/Pathfinding/VillagerWaypointsTraverser.cs

```diff
@@ -282,28 +282,25 @@ public override void OnGameTick(float dt)
             }
         }
 
-        private bool handleDoors(Vec3d target, string state)
+        private bool CanBeOpenedOrClosed(BlockPos pos, string state)
         {
-            var targetBlock = entity.World.BlockAccessor.GetBlock(target.XInt, target.YInt, target.ZInt) as BlockBaseDoor;
-            if (targetBlock != null && targetBlock.Variant["state"] == state)
+            var block = entity.World.BlockAccessor.GetBlock(pos);
+            if (block is BlockBaseDoor && block.Variant["state"] == state)
             {
-                entity.World.PlaySoundAt(AssetLocation.Create(targetBlock.Attributes["triggerSound"].AsString("sounds/block/door"), targetBlock.Code.Domain), target.X + 0.5f, target.Y + 0.5f, target.Z + 0.5f);
-                entity.World.BlockAccessor.WalkBlocks(new BlockPos(target.XInt - 1, target.YInt - 1, target.ZInt - 1), new BlockPos(target.XInt + 1, target.YInt + 1, target.ZInt + 1), (block, x, y, z) => handleDoor(block, new BlockPos(x, y, z)));
-
+                var minPos = new BlockPos(target.XInt - 1, target.YInt - 1, target.ZInt - 1);
+                var maxPos = new BlockPos(target.XInt + 1, target.YInt + 1, target.ZInt + 1);
+                entity.World.BlockAccessor.WalkBlocks(minPos, maxPos, (onBlock, x, y, z) => OpenOrClose(new BlockPos(x, y, z), state));
```

Comment:
does it also work if you dont do the walkblocks to check surrounding blocks but instead do `OpenOrClose(target.ToBlockPos())` only on the targetblock itself?

Reply:
I haven't tried that